### PR TITLE
doc(partner_sdk): Alter Table operation guidelines

### PR DESCRIPTION
**Pull request by fivetran-rishabhghosh**
Links https://fivetran.height.app/T-797932/description
Add guidelines for Alter Table operation

### CODE REVIEW

**Review by varundhall — APPROVED** [review]

**Review by shivangisingh172 — APPROVED** [review]

File: development-guide.md [hunk 1]
```diff
@@ -125,6 +125,10 @@ This operation should report all columns in the destination table, including Fiv
 - This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.
 - `utc_delete_before` has millisecond precision.
 
+#### AlterTable
+- This operation should exercise change of primary key columns, addition of columns, change of data types, etc.
```
Comment (fivetran-dejantucakov) [inline development-guide.md:129]: @fivetran-rishabhghosh Ending with etc. is very vague. Can you provide more specific operations?

Comment (fivetran-dejantucakov) [inline development-guide.md:129]:
```suggestion
- This operation should perform changes to primary key columns and data types, column additions, (INSERT OTHER)
```

File: development-guide.md [hunk 2]
```diff
@@ -125,6 +125,10 @@ This operation should report all columns in the destination table, including Fiv
 - This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.
 - `utc_delete_before` has millisecond precision.
 
+#### AlterTable
+- This operation should exercise change of primary key columns, addition of columns, change of data types, etc. 
+- However, this operation should not drop any columns even if the alter table request has a table with a different set of columns. Dropping of columns could lead to unexpected data loss for customers and is against Fivetran's general approach to data movement.
```
Comment (fivetran-dejantucakov) [inline development-guide.md:130]:
```suggestion
- However, this operation should not drop any columns even if the AlterTable request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
```

**Review by fivetran-dejantucakov — CHANGES_REQUESTED** [review]
@fivetran-rishabhghosh Please see suggestions and comments.

**Comment by fivetran-rishabhghosh** [comment]
@fivetran-dejantucakov Updated

File: development-guide.md [hunk 3]
```diff
@@ -125,6 +125,10 @@ This operation should report all columns in the destination table, including Fiv
 - This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.
 - `utc_delete_before` has millisecond precision.
 
+#### AlterTable
+- This operation should exercise change of primary key columns, addition of columns and change of data types. 
+- However, this operation should not drop any columns even if the AlterTable request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
```
Comment (fivetran-dejantucakov) [inline development-guide.md:130]:
```suggestion
- This operation should be used for changing primary key columns, adding columns, and changing data types. 
- However, this operation should not drop any columns even if the `AlterTable` request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
```

**Review by fivetran-dejantucakov — CHANGES_REQUESTED** [review]
@fivetran-rishabhghosh Please see suggestions.

**Review by fivetran-dejantucakov — APPROVED** [review]
@fivetran-rishabhghosh Approved, tnx!

File: development-guide.md [hunk 4]
```diff
@@ -125,6 +125,10 @@ This operation should report all columns in the destination table, including Fiv
 - This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.
 - `utc_delete_before` has millisecond precision.
 
+#### AlterTable
+- This operation should be used for changing primary key columns, adding columns, and changing data types. 
+- However, this operation should not drop any columns even if the `AlterTable` request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
```
Comment (alsnkline) [inline development-guide.md:130]: This doesn't say what they should do. i'd also love to link to a doc that describes our policy/what we usually do - ie not dropping columns/deleting data - but I didn't just find a suitable link.

eg 'this operation should not drop any columns even if the 'AlterTabel' request has a table with a different set of columns. Fivetran's approach to data movement is to leave such columns in a customer's warehouse with no additional data populating to provide the customer control and choice on timing for removal. Dropping columns can lead to unexpected customer data loss.

**Pull request merged**